pkg/v1/cli/artifact/local.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package artifact
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/common"
+)
+
+// LocalArtifact defines local artifact path
+type LocalArtifact struct {
+	Path string
+}
+
+// NewLocalArtifact creates Local Artifact object
+func NewLocalArtifact(path string) Artifact {
+	// If path is not an absolute path
+	// search under `xdg.ConfigHome/tanzu-plugin/localPath` directory
```

Comment:
**nitpick**: It seems like we allow both absolute and relative paths, and that there's some important information about how a relative path is handled. I'd capture all of that in the method-level comment so that someone looking at the godoc for the method can easily see it (without having to click into the method definition).